### PR TITLE
[pgeo] Add support for field domains

### PR DESCRIPTION
**Pull request by nyalldawson**

### CODE REVIEW

File: gdal/ogr/ogrsf_frmts/pgeo/ogrpgeotablelayer.cpp [hunk 1]
```diff
@@ -195,6 +195,63 @@ CPLErr OGRPGeoTableLayer::Initialize( const char *pszTableName,
 
     poFeatureDefn->SetGeomType( eOGRType );
 
+    // try to associate domains with fields
+    CPLODBCStatement oTableList( poSession );
```
Comment (rouault) [inline gdal/ogr/ogrsf_frmts/pgeo/ogrpgeotablelayer.cpp:199]: as we have already done that in OGRPGeoDataSource::Open(), couldn't we just have a OGRPGeoDataSource::m_bHasGdbItemsTable member and test it here ?

File: gdal/ogr/ogrsf_frmts/pgeo/ogrpgeotablelayer.cpp [hunk 2]
```diff
@@ -195,6 +195,63 @@ CPLErr OGRPGeoTableLayer::Initialize( const char *pszTableName,
...
+                        if ( const CPLXMLNode* psFieldInfoExs = CPLGetXMLNode(oTree.get(), "=DEFeatureClassInfo.GPFieldInfoExs") )
+                        {
+                            for( const CPLXMLNode *psFieldInfoEx =
+                                     CPLGetXMLNode( psFieldInfoExs, "GPFieldInfoEx" );
+                                 psFieldInfoEx != nullptr;
+                                 psFieldInfoEx = psFieldInfoEx->psNext )
+                            {
+                                const CPLString osName = CPLGetXMLValue(psFieldInfoEx, "Name", "");
+                                const CPLString osDomainName = CPLGetXMLValue(psFieldInfoEx, "DomainName", "");
+                                if ( osDomainName != "" )
```
Comment (rouault) [inline gdal/ogr/ogrsf_frmts/pgeo/ogrpgeotablelayer.cpp:240]:
```suggestion
                                if ( !osDomainName.empty() )
```

File: gdal/ogr/ogrsf_frmts/pgeo/ogrpgeotablelayer.cpp (hunk 1 above)
Comment (nyalldawson) [reply gdal/ogr/ogrsf_frmts/pgeo/ogrpgeotablelayer.cpp:199]: That was my original approach, but since the layers are created by the datasource BEFORE we iterate over the tables list we haven't yet performed that table iteration at the time this code is called...

Comment (rouault) [reply gdal/ogr/ogrsf_frmts/pgeo/ogrpgeotablelayer.cpp:199]: looking a bit at the Open() code, couldn't we slighly modify it to:
- list all table name and store them in a set + set m_bHasGdbItemsTable
- instantiate spatial tables (unchanged)
- instantiate non-spatial tables (modified to iterate over the collected set of names)

Comment (nyalldawson) [reply gdal/ogr/ogrsf_frmts/pgeo/ogrpgeotablelayer.cpp:199]: Sure, I can do that

Comment (nyalldawson) [reply gdal/ogr/ogrsf_frmts/pgeo/ogrpgeotablelayer.cpp]: Ok, done

**Comment by rouault** [comment]
Conflict to solve

**Comment by nyalldawson** [comment]
@rouault fixed

**Comment by rouault** [comment]
```
================================= FAILURES ===================================
__________________________ test_ogr_pgeo_read_domains __________________________
    def test_ogr_pgeo_read_domains():

>       ds = gdal.OpenEx('data/pgeo/domains.mdb', gdal.OF_VECTOR)
E       NameError: name 'gdal' is not defined
```

File: autotest/ogr/ogr_pgeo.py [hunk 3]
```diff
@@ -366,3 +367,43 @@ def test_ogr_pgeo_11():
                                       max_error=0.0001) != 0:
         feat.DumpReadable()
         pytest.fail('did not get expected geometry')
+
+###############################################################################
+
+
+def test_ogr_pgeo_read_domains():
+
+    ds = gdal.OpenEx('data/pgeo/domains.mdb', gdal.OF_VECTOR)
```
Comment (rouault) [inline autotest/ogr/ogr_pgeo.py:376]: you likely need a test of ds is not None like in test_ogr_pgeo_11() to detect misconfigured driver tof fix the test failures
